src/main/java/edu/gatech/cs2340/trydent/Audio.java

```diff
@@ -0,0 +1,190 @@
+     * @return a builder object for creating sound effects
+     */
+    public static SoundEffectBuilder createSoundEffect(String soundEffectFilename) {
+        return new SoundEffectBuilder(soundEffectFilename);
+    }
+
+    /**
+     * Sets the master volume, which affects both music and sound effects
+     * Volume is calculated by MasterVolumn * IndividualVolumn
+     * @param newMasterVolume the new master volume (should be between 0.0 and 1.0)
+     */
+    public static void setMasterVolume(double newMasterVolume){
+        masterVolume = newMasterVolume;
+        if(music != null) {
+            music.setVolume(masterVolume * musicMolume);
+        }
+        for(MediaContainer soundEffect : soundEffects) {
+            soundEffect.player.setVolume(masterVolume * soundEffect.playerVolume);
+        }
+    }
+
+    /**
+     * Sets the volume for music.
+     * This value is always multiplied by the master volume,
+     * so the resulting volumne is MasterVolume * new_music
+     * @param newMusicVolume the new music volume
+     */
+    public static void setMusicVolume(double newMusicVolume){
+        musicMolume = newMusicVolume;
+        if(music != null) {
+            music.setVolume(masterVolume * musicMolume);
+        }
+    }
+
+    /**
+     * Pauses all of the audio.
+     * This method is useful for when the game is paused or is minimized.
+     */
+    public static void pauseAudio() {
+        playing = false;
+        if(music != null) {
+            music.pause();
+        }
+        for(MediaContainer soundEffect : soundEffects) {
+            soundEffect.player.pause();
+        }
+    }
+
+    /**
+     * Resumes audio or starts playing.
+     */
+    public static void resumeAudio() {
+        playing = true;
+        if(music != null) {
+            music.play();
+        }
+        for(MediaContainer soundEffect : soundEffects){
+            soundEffect.player.play();
+        }
+    }
+
+    /**
+     * Determines if the audio is currently playing.
+     * @return true if currently playing
+     */
+    public static boolean isPlaying(){
+        return playing;
+    }
+
+    /**
+     * A builder class for creating sound effects.
+     */
+    public static class SoundEffectBuilder {
+        private MediaPlayer soundEffect;
+        private double volume;
+        private Runnable callback;
+
+        private SoundEffectBuilder(String soundEffectFilename) {
+            soundEffect = new MediaPlayer(new Media(soundEffectFilename));
+            soundEffect.setCycleCount(1);
+            soundEffect.setVolume(masterVolume);
+            callback = () -> {};
+        }
+
+        /**
+         * This method sets the balance for a sound effect.
+         * This can be used to add a sense of direction to sound.
+         * @param balance a value between -1.0 (left) to 1.0 (right)
+         * @return current sound effect being built
+         */
+        public SoundEffectBuilder setBalance(double balance){
```

Comment:
For whatever reason, our checkstyle does not appear to catch this issue. Anyways, I'll grep for this and fix this.